website/content/v1.10/advanced/kernel-module.md

```diff
@@ -0,0 +1,260 @@
+---
+title: "Adding a Kernel Module"
+description: "Create a system extension that includes kernel modules."
+---
+
+[System extensions]({{< relref "../talos-guides/configuration/system-extensions" >}}) in Talos provide ways to add files to the root filesystem and make it possible to run privileged containers as services.
+But Talos still requires that kernel modules be signed with a trusted signing key to be loaded at run time.
+
+To add a kernel module to Talos you will need to create a system extension that is built with the kernel and signed by the same signing key.
+System extensions without kernel modules work without this requirement, but there are a couple extra steps needed when adding kernel modules.
+
+## Create a package
+
+Talos is built from the [pkgs](https://github.com/siderolabs/pkgs/) repo and the first step will be to add your custom package to that repo to be built with Talos.
+You can look at other packages in that repository for examples of what should be included.
+
+The only file required to create a package is a pkg.yaml file in a folder.
+Let's create an example package to walk through each step.
+
+Clone the repo and create a folder.
+
+```bash
+git clone https://github.com/siderolabs/pkgs
+cd pkgs
+mkdir my-module
+```
+
+Now add the package to the `.kres.yaml` file in the root of the repository.
+We use this file for templating and generating Makefiles.
+
+```yaml
+ spec:
+   targets:
+     ...
+     - my-module
+     ...
+```
```

Comment:
`make rekres` step missing

Reply:
and I think we have a separate list of kernel-dependent targets